### ISSUE TITLE
Visual studio code intellisense lags while trying to create a Schema

### ISSUE THREAD

**Issue by perenSHERE** [Automattic/mongoose#10492]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
Whenever I try to create a Schema as new mongoose.Schema() it lags the intellisense and if I require the schema it lags the whole project intellisense
**If the current behavior is a bug, please provide the steps to reproduce.**
```
const {Schema} = require('mongoose')
const exampleSchma = new Schema(schemaConfig)
module.exports = exampleSchema;
```
This makes the intellisense inside the file super slow
trying to require the file lags the whole project

**What is the expected behavior?**
The intellisense should respond fastly
**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node =14.17.3
Mongoose = 5.13.3
Mongodb = 5.0.1

**The extensions I am using are prettier and babel javascript**

**Comment by hanatic** [comment]
I sometimes get this too: it's a problem with VSCode not Mongoose though. Report it to the VSCode repo please.

**Comment by HamoBoker** [comment]
Same thing with webstorm

Node 16.5.0
Mongoose 5.13.3

**Comment by perenSHERE** [comment]
> I sometimes get this too: it's a problem with VSCode not Mongoose though. Report it to the VSCode repo please.

No, it's a specific problem with the mongoose latest versions I tried using 4.13.20 version of mongoose the intellisense works fine but as soon as I try any 5.0 or above versions it makes the intellisense slow.

**Comment by IslandRhythms** [comment]
Link the extension you are using. visualstudioexptteam.vscodeintellicode is the one I am using and it works fine

**Comment by mjfwebb** [comment]
Fairly sure this is related https://github.com/Automattic/mongoose/issues/10349

**Comment by perenSHERE** [comment]
The extensions I am using are prettier and babel javascript

**Comment by perenSHERE** [comment]
> This issue is a replication of #10349
I m closing the issue
> This is a problem related to mongoose types and they should for sure fix it.
**"I think this may need a different generics system, possibly recoding from scratch the declarations for the Schema, with more than likely backwards breaking changes"**
Shadiest goat
>for a temporary fix refer too #10349

**Comment by AdsonCicilioti** [comment]
Intellisens still very slow. :(
v.6.6.2
